addons/l10n_latam_check/models/account_payment.py

```diff
@@ -0,0 +1,379 @@
+import stdnum
+from odoo import fields, models, _, api
+from odoo.exceptions import UserError, ValidationError
+from odoo.tools.misc import format_date
+from odoo.osv import expression
+import logging
+_logger = logging.getLogger(__name__)
+
+
+class AccountPayment(models.Model):
+    _inherit = 'account.payment'
+
+    # Third party check operation links
+    l10n_latam_check_id = fields.Many2one(
+        comodel_name='account.payment',
+        string='Check',
+        readonly=True, states={'draft': [('readonly', False)]},
+        copy=False,
+    )
+    l10n_latam_check_operation_ids = fields.One2many(
+        comodel_name='account.payment',
+        inverse_name='l10n_latam_check_id',
+        string='Check Operations',
+        readonly=True,
+    )
+    l10n_latam_check_current_journal_id = fields.Many2one(
+        comodel_name='account.journal',
+        string="Check Current Journal",
+        compute='_compute_l10n_latam_check_current_journal', store=True,
+    )
```

Comment:
When is the journal different than the one of the payment?

Reply:
This is on the original payment, so afterwards you can have used the check to pay a customer and then it could have come back and e.g. have been put on a rejected checks journal. (so it goes from one to the other)  (if it goes to the vndor, it should be empty)